### PR TITLE
Replace pre-commit with prek

### PR DESCRIPTION
**Pull request by weiji14**
[Prek](https://prek.j178.dev/) is a drop-in replacement of pre-commit, that aligns with our hip usage of `uv`, `ty`, and other Rust-based tooling!

Test using:

```
uv run prek install --install-hooks
uv run prek run --all-files
```

Also fixing some pre-commit errors seen in https://github.com/developmentseed/pixelverse/actions/runs/20699116381/job/59418754998, and some new ones after updating some pre-commit hooks.

<details>

```python-traceback
error[invalid-return-type]: Return type does not match returned value
  --> src/pixelverse/download/sentinel1.py:72:12
   |
70 |     da_linear = dataarray.where(cond=dataarray != 0)
71 |     da_decibel = 10 * np.log10(da_linear)
72 |     return da_decibel
   |            ^^^^^^^^^^ expected `DataArray`, found `ndarray[tuple[Any, ...], Unknown]`
   |
  ::: src/pixelverse/download/sentinel1.py:56:51
   |
56 | def linear_to_decibel(dataarray: xr.DataArray) -> xr.DataArray:
   |                                                   ------------ Expected `DataArray` because of return type
57 |     """Transform VV or VH values from linear to decibel scale.
   |
info: rule `invalid-return-type` is enabled by default

error[invalid-method-override]: Invalid override of method `forward`
  --> src/pixelverse/models/transforms.py:7:9
   |
 6 | class PixelTimeSeriesNormalize(T.Normalize):
 7 |     def forward(self, tensor: torch.Tensor) -> torch.Tensor:
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Definition is incompatible with `Transform.forward`
 8 |         assert tensor.ndim in [2, 3], (
 9 |             "Input must be a 2D (time, channels) or 3D (batch, time, channels) tensor"
   |
  ::: .venv/lib/python3.13/site-packages/torchvision/transforms/v2/_transform.py:57:9
   |
55 |         raise NotImplementedError
56 |
57 |     def forward(self, *inputs: Any) -> Any:
   |         ---------------------------------- `Transform.forward` defined here
58 |         """Do not override this! Use ``transform()`` instead."""
59 |         flat_inputs, spec = tree_flatten(inputs if len(inputs) > 1 else inputs[0])
   |
info: This violates the Liskov Substitution Principle
info: rule `invalid-method-override` is enabled by default

Found 2 diagnostics
```

</details>

References:
- https://prek.j178.dev/quickstart/

### CODE REVIEW

File: src/pixelverse/models/transforms.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import torch
-import torchvision.transforms.v2 as T
+import torchvision.transforms as T
```
Comment (weiji14) [inline src/pixelverse/models/transforms.py]: We were trying to subclass from torchvision.transforms.v2.Normalize, specifically the `forward` method, but this is explicitly discouraged at https://github.com/pytorch/vision/blob/main/torchvision/transforms/v2/_transform.py#L57-L58 (it is [recommended to override the `transform` method instead](https://docs.pytorch.org/vision/master/auto_examples/transforms/plot_custom_transforms.html#basics-override-the-transform-method)).

I'm actually a little unsure if the normalization was applied correctly before this, and a little confused/worried on why we are importing from torchvision's private API (`torchvision.models._api`) here:

https://github.com/developmentseed/pixelverse/blob/a1498e150ecbcfff09daf31196bc78cf7e330959/src/pixelverse/models/tessera.py#L6

@isaaccorley, can you maybe double check whether the mean/std normalization has been applied correctly? I see the original Tessera code at https://github.com/ucam-eo/tessera/blob/e26a1f05a4d2cb45510eca1aa65d585899ef891e/tessera_infer/src/multi_tile_infer.py#L190-L191 appears to be doing the normalization as a pre-processing step (i.e. in the Dataset/Dataloader) rather than in the model as it appears to be happening here in pixelverse. These two lines look magic to me:

https://github.com/developmentseed/pixelverse/blob/a1498e150ecbcfff09daf31196bc78cf7e330959/src/pixelverse/models/tessera.py#L279-L280

Have you added extra transform layers to the model weights at https://huggingface.co/isaaccorley/tessera? A little hard to tell where the mean/std normalization is being applied, or if it is being applied at all.

**Review by isaaccorley — CHANGES_REQUESTED** [review]

File: src/pixelverse/models/transforms.py (hunk 1 above)
Comment (weiji14) [reply src/pixelverse/models/transforms.py]: > 3. I haven’t added an extra transform layer to the model. The normalization would need to be applied before passing data to the model like this:
> 
> 
> ```python
> import pixelverse as pv
> 
> model, transform = create_model(“tessera”)
> embeddings = model(transform(x))
> ```

Well if there's no transform layer, then we're probably not applying normalization in the current `generate_embeddings` function:

https://github.com/developmentseed/pixelverse/blob/a1498e150ecbcfff09daf31196bc78cf7e330959/src/pixelverse/generate_embeddings.py#L40-L66

I'm seeing that we're essentially doing `embeddings = model(x)` only. So does that mean adding the mean/std weights in `class TESSERA_WEIGHTS(WeightsEnum) -> TESSERA = Weights(..., transforms=tessera_transforms, ...)` doesn't do anything?

Comment (weiji14) [reply src/pixelverse/models/transforms.py]: Ah, but in `test_models.py`, we do apply the transform:

https://github.com/developmentseed/pixelverse/blob/a1498e150ecbcfff09daf31196bc78cf7e330959/tests/test_models.py#L8-L16

So we just need to update the `generate_embeddings` function to do the same. Let me do that in a separate PR.

File: pyproject.toml [hunk 2]
```diff
@@ -122,6 +126,5 @@ checks = [
   "RT03",
 ]
 
-[tool.ty.src]
-include = [ "src", "tests" ]
-exclude = [ "notebooks" ]
+[tool.typos.default.extend-words]
+inpt = "inpt"
```
[the remark this answers is not in the document]
Comment (weiji14) [reply pyproject.toml]: inpt has to be used to conform to torchvision's method.
